tests/BoundaryClickWatcher/BoundaryClickWatcher.js

```diff
@@ -0,0 +1,35 @@
+import { React, expect, shallow, jsDomGlobal } from '../setup';
+import BoundaryClickWatcher from '../../lib/BoundaryClickWatcher';
+
+let wrapper;
+
+describe('BoundaryClickWatcher', () => {
+  jsDomGlobal();
+
+  beforeEach(() => {
+    wrapper = shallow(
+      <BoundaryClickWatcher>
+        {(boundaryIsActive) => {
+          return <div isActive={boundaryIsActive} />;
+        }}
+      </BoundaryClickWatcher>
+    );
+  });
+
+  it('boundaryIsActive is true on click', () => {
+    expect(wrapper.state('boundaryIsActive')).to.equal(false);
+    wrapper.simulate('click');
+    expect(wrapper.state('boundaryIsActive')).to.equal(true);
+  });
+
+  it('passes boundaryIsActive to child', () => {
```

Comment:
Did you want to test when a child is passed that isn't a function? If you can't imagine a use case for this then maybe we should expect a function to be passed as the child 👍